database.go

```diff
@@ -1,37 +1,81 @@
 package main
 
 import (
-	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
 )
 
-var db *sql.DB
+var (
+	ErrDBConnStrParseNoSep = errors.New("Missing ':' in connection string")
+	ErrDBConnStrParseNotImplemented = func(dbName string) error {
```

Comment:
Can we have consistently use functions to return/generate errors? Instead of some being functions and some being regular `error` objects.